torch/fx/passes/backends/nvfuser/compile.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+
+import torch
+from torch.fx import GraphModule
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.backends.nvfuser.operator_support import NvFuserOperatorSupport
+from torch._prims.executor import execute
+from torch.fx.experimental.proxy_tensor import DecompositionInterpreter
+from torch._decomp import decomposition_table
+
+import logging
+
+logging.basicConfig(level=logging.WARNING)
+logger = logging.getLogger(__name__)
+
+def aten_to_dtype(self, dtype: torch.dtype, **kwargs):
+    if len(kwargs) > 0 or not dtype:
+        raise RuntimeError("No support for other to.dtype() formats other than to.dtype(self, dtype)")
+    return torch._prims.convert_element_type(self, dtype)
+
+# decomposition_table currently contains both aten2aten and aten2prim decomposition
+# this is a hack to seperate them, as we only need aten2prim decomposition for nvfuser-supported aten graph lowering
+aten2aten_decomp = {}
+aten2prim_decomp = {}
+
+for op, decomp_fn in decomposition_table.items():
+    if "torch._refs" in decomp_fn.__module__:
+        aten2prim_decomp[op] = decomp_fn
+    else:
+        aten2aten_decomp[op] = decomp_fn
+
+aten2prim_decomp[torch.ops.aten.to.dtype] = aten_to_dtype
+
+class NvFuserBackend:
+    def __init__(self):
+        self.supported_ops = NvFuserOperatorSupport()
```

Comment:
This doesn't actually have any mutable state so it would be clearer if this was just a singleton object (actually, it's not really clear to me why operator supports need to be a class in the first place, it's literally just a callable without any state.)

Reply:
OperatorSupport was implemented as an interface class in the first place. NvFuserOperatorSupport is a subclass for OperatorSupport.

Reply:
OK, updated prescription (but it doesn't have to be done in this PR): I think OperatorSupport should be represented just as a Callable, and you're welcome to pass in a function or a proper class that implements `__call__`. To represent this in typing you can write an alias for `Callable[...]`.

This is Python, not Java; we don't have to create an interface class just for a function, we can just pass in a function directly.

Reply:
OK, I will update OperatorSupport in the next PR.